### PR TITLE
Adds machinery process mult

### PR DESCRIPTION
**Pull request by Superlagg**
[FEATURE]
## About the PR 

Adds in a proc that I stole from the lifeloop that'll set a multiplier if the time between processes is more than what's expected. Should properly compensate for the fact that different machine tiers process at different rates.

I also added this to the recharger, so it should charge more if the server's lagging.

## Why's this needed? 

Gotta murder greytide fast and server too slow.

## Changelog

```
(u)Superlagg:
(+)Rechargers and manufacturers should no longer take ages to work when exposed to spacelag.
(+)Manufacturers should now deliver the thing you want it to make in the amount of time it promises.
```

### CODE REVIEW

File: code/obj/machinery.dm [hunk 1]
```diff
@@ -23,7 +23,12 @@
 	var/tmp/processing_tier = PROCESSING_EIGHTH
 	var/tmp/current_processing_tier
 	var/tmp/machine_registry_idx // List index for misc. machines registry, used in loops where machines of a specific type are needed
+	var/const/base_tick_spacing = 10 // Machines proc every 1*(2^tier-1) seconds. Or something like that.
+	var/const/cap_base_tick_spacing = 300 //
```
Comment (Adharainspace) [inline code/obj/machinery.dm:27]: i wouldnt set these to const if i were you. being able to adjust these in game is good for debugging. and const vars dont really save on perf at all.
```suggestion
	var/base_tick_spacing = 10 // Machines proc every 1*(2^tier-1) seconds. Or something like that.
	var/cap_base_tick_spacing = 300 //
```

Comment (Adharainspace) [reply code/obj/machinery.dm:27]: also you dont say anything with this comment

File: code/obj/machinery/recharger.dm [hunk 2]
```diff
@@ -157,22 +157,24 @@ obj/machinery/recharger
 	// WHO DID THIS
 	// WHY DID YOU DO THIS
 	// die
+	last_process = TIME
```
Comment (Adharainspace) [inline code/obj/machinery/recharger.dm:160]: please move the last_process thing to the base machinery loop or loop(s) if you can! having to re-set that manually when you could just call your parents seems kinda bad.

Comment (Superlagg) [reply code/obj/machinery/recharger.dm]: Okay, did that!

**Comment by Superlagg** [comment]
Okay now, the mult tends to be 10 without lag and a lot more sometimes for some reason. Huh.

**Comment by Superlagg** [comment]
Okay turns out this is fricked beyond even pretending it works. Closing this until I unfrick it.

**Comment by Superlagg** [comment]
okay turns out i fricked it by trying to use ^ as an exponent. it works now.

also I added it to manufacturers. seems to work.

**Comment by Superlagg** [comment]
Okay, I moved everything I could to the machine process loop, now it should be nice and easy to add mults to any other machine.

Also rechargers might charge _slightly_ faster than they used to. Like, 2-3 per charge, roughly. Its cus I had to kinda eyeball the base tick rate since it varies quite a bit even without lag.

File: code/obj/machinery.dm (hunk 1 above)
Comment (ZeWaka) [reply code/obj/machinery.dm]: I mean, const variables /are/ faster

**Comment by ZeWaka** [comment]
this is definitely not a major change imo

File: code/datums/controllers/process/machines.dm [hunk 3]
```diff
@@ -68,14 +69,17 @@ datum/controller/process/machines
...
+
 			for(var/X in machlist[(src.ticker % (1<<(i-1)))+1])
 				if(!X) continue
 				var/obj/machinery/machine = X
 				if( machine.z == 4 && !Z4_ACTIVE ) continue
 		#ifdef MACHINE_PROCESSING_DEBUG
 				var/t = world.time
 		#endif
-				machine.process()
+				mult = clamp(TIME - machine.last_process, machine.base_tick_spacing*(2**(machine.processing_tier-1)), machine.cap_base_tick_spacing*(2**(machine.processing_tier-1))) / (machine.base_tick_spacing*(2**(machine.processing_tier-1)))
```
Comment (pali6) [inline code/datums/controllers/process/machines.dm:80]: It'd be much more readable if you put the target machine loop time into a separate var first as it repeats three times in the expression

Comment (Superlagg) [reply code/datums/controllers/process/machines.dm]: Okay I moved the base and cap vars to vars in the loop. Should they be defined every loop like this, or defined at the top and reassigned every loop?

**Review by pali6 — APPROVED** [review]

**Pull request merged**